### PR TITLE
ams identity enabled

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Azure Media Service Managed Identity Enabled',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:5]: Media Services Managed Identity Enabled

Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:5]: Media Services Managed Identity Enabled

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    category: 'Media Services',
+    domain: 'Identity and Access Management',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:7]: update domain

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    helpers.addResult(results, 3, `Unable to query for Media Service: ${helpers.addError(getMediaService)}`,
+                        location, mediaService.id);
+                    continue;
+                }
+
+                if (getMediaService.data.identity && getMediaService.data.identity.type 
+                && (getMediaService.data.identity.type.toLowerCase() === 'userassigned' ||
+                 getMediaService.data.identity.type.toLowerCase() === 'systemassigned')) {
+
+                    helpers.addResult(results, 0, 'Managed Identity is enabled for Azure Media Service account', location, mediaService.id);
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:51]: Update the message to **Media Service account has managed Identity enabled**

File: helpers/azure/api.js [hunk 4]
```diff
@@ -910,6 +915,13 @@ var postcalls = {
             properties: ['id'],
             url: 'https://management.azure.com/{id}/configurations?api-version=2017-12-01'
         }
+    },
+    mediaServices: {
```
Comment (alphadev4) [inline helpers/azure/api.js:919]: we can simply use listAll

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Media Services Managed Identity Enabled',
+    category: 'Media Services',
+    domain: 'Content Delivery',
+    description: 'Ensure that Azure Media Services have managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:8]:
```suggestion
    description: 'Ensures that Azure Media Services have managed identity enabled.',
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    more_info: 'Enabling managed identities eliminate the need for developers having to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.',
+    link: 'https://learn.microsoft.com/en-us/azure/media-services/latest/concept-managed-identities',
+    recommended_action: 'Remove Azure Media Services accounts and create a new account with managed identity enabled.',
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:11]:
```suggestion
    recommended_action: 'Create a new Media service account with managed identity enabled.',
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            }
+
+            for (let mediaService of mediaServices.data) {
+                if (!mediaService.id) continue;
+
+                var getMediaService = helpers.addSource(cache, source, 
+                    ['mediaServices', 'get', location, mediaService.id]);
+
+                if (!getMediaService || getMediaService.err || !getMediaService.data) {
+                    helpers.addResult(results, 3, `Unable to query for Media Service: ${helpers.addError(getMediaService)}`,
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:42]:
```suggestion
                    helpers.addResult(results, 3, `Unable to query for Media Service data: ${helpers.addError(getMediaService)}`,
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+                        location, mediaService.id);
+                    continue;
+                }
+
+                if (getMediaService.data.identity && getMediaService.data.identity.type 
+                && (getMediaService.data.identity.type.toLowerCase() === 'userassigned' ||
+                 getMediaService.data.identity.type.toLowerCase() === 'systemassigned')) {
+
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:50]:
```suggestion
                if (getMediaService.data.identity && getMediaService.data.identity.type  &&
                   (getMediaService.data.identity.type.toLowerCase() === 'userassigned' ||
                    getMediaService.data.identity.type.toLowerCase() === 'systemassigned')) {

```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+                    helpers.addResult(results, 0, 'Media Service account has managed Identity enabled', location, mediaService.id);
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:51]:
```suggestion
                    helpers.addResult(results, 0, 'Media Service account has managed identity enabled', location, mediaService.id);
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Media Service account does not have managed Identity enabled', location, mediaService.id);
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.js:53]:
```suggestion
                    helpers.addResult(results, 2, 'Media Service account does not have managed identity enabled', location, mediaService.id);
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.spec.js [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+                done();
+            });
+        });
+
+        it('should give passing result if managed identity is not enabled', function(done) {
+            const cache = createCache([mediaServices[0]], getMediaService[0]);
+            amsManagedIdentityEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
+                expect(results[0].message).to.include('Media Service account does not have managed Identity enabled');
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.spec.js:103]:
```suggestion
                expect(results[0].message).to.include('Media Service account does not have managed identity enabled');
```

File: plugins/azure/mediaServices/amsManagedIdentityEnabled.spec.js [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+                done();
+            });
+        });
+
+        it('should give failing result if classic API enabled', function(done) {
+            const cache = createCache([mediaServices[0]], getMediaService[1]);
+            amsManagedIdentityEnabled.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].message).to.include('Media Service account has managed Identity enabled');
```
Comment (alphadev4) [inline plugins/azure/mediaServices/amsManagedIdentityEnabled.spec.js:114]:
```suggestion
                expect(results[0].message).to.include('Media Service account has managed identity enabled');
```

**Review by alphadev4 — APPROVED** [review]

**Review by mehakseedat63 — APPROVED** [review]

**Pull request merged**